js/foam/oauth2/GoogleSignIn.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * @license
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+CLASS({
+  package: 'foam.oauth2',
+  name: 'GoogleSignIn',
+  requires: [
+    'foam.ui.md.PopupView',
+  ],
+  imports: [
+    'document',
+    'googleClientId',
+    'window',
+  ],
+
+  constants: {
+    COOKIE_KEY: 'foamDAOAuth',
```

Comment:
Upon further thought, I don't actually like using a cookie for this.

You can't set the cookies scope using document.cookie like you can when a server sets it.  So I'm not confident that it won't send the cookie for requests to other domains or non https requests.

How about we use an extra header in our XHRs.  X-FOAM-Auth.  Then we can decorate the XHRs for a client dao like we already do for OAuth2.

Plus then we can contain the auth token in our XHR decorator so that no other part of the app has access, and we can enforce that we only send it to domains we trust.  Like only https to our apps server, etc, etc.  Makes it harder for the client to screw up.